### PR TITLE
Fix a bug in network telemetry manager

### PR DESCRIPTION
**Pull request by suryanshup**
-- Enable the polling timer only when the device is
   network provisioined.

### CODE REVIEW

File: src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h [hunk 1]
```diff
@@ -103,7 +103,7 @@
  * The maximum size (in bytes) of a debug logging message.
  */
 #ifndef WEAVE_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE
-#define WEAVE_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE 256
+#define WEAVE_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE 512
```
Comment (jaylogue) [inline src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h]: Why is this needed for all devices?

File: src/adaptations/device-layer/NetworkTelemetryManager.cpp [hunk 2]
```diff
@@ -109,6 +112,39 @@ WEAVE_ERROR NetworkTelemetryManager::Init(void)
...
 }
 
+void NetworkTelemetryManager::OnPlatformEvent(const WeaveDeviceEvent* event)
+{
+#if WEAVE_DEVICE_CONFIG_ENABLE_WIFI_TELEMETRY
+    if (event->Type == DeviceEventType::kWiFiConnectivityChange)
+    {
+        if (ConnectivityMgr().IsWiFiStationProvisioned() && ConnectivityMgr().IsWiFiStationEnabled())
+        {
+            mWiFiTelemetry.Enable();
```
Comment (jaylogue) [inline src/adaptations/device-layer/NetworkTelemetryManager.cpp]: This has the effect of forcing telemetry on whenever WiFi connectivity changes.  Is that the correct thing to do in every case?  It effectively means the pubic Disable() function is useless, because it wont stay disabled for long.

Comment (suryanshup) [reply src/adaptations/device-layer/NetworkTelemetryManager.cpp]: Is there a use case for disabling telemetry once it has been enabled ?

File: src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h (hunk 1 above)
Comment (suryanshup) [reply src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h]: I'll move this change to 
src/adaptations/device-layer/include/Weave/DeviceLayer/nRF5/WeaveDevicePlatformConfig.h
instead.

Comment (suryanshup) [reply src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h]: Moved this to the weave config file in the lock example app.
PR is here: https://github.com/openweave/openweave-nrf52840-lock-example/pull/57

File: src/adaptations/device-layer/NetworkTelemetryManager.cpp (hunk 2 above)
Comment (suryanshup) [reply src/adaptations/device-layer/NetworkTelemetryManager.cpp]: Fixed

File: src/adaptations/device-layer/NetworkTelemetryManager.cpp [hunk 3]
```diff
@@ -109,14 +110,18 @@ WEAVE_ERROR NetworkTelemetryManager::Init(void)
...
 #if WEAVE_DEVICE_CONFIG_ENABLE_WIFI_TELEMETRY
 void WiFiTelemetry::GetTelemetryStatsAndLogEvent(void)
 {
     WEAVE_ERROR err;
 
-    err = ConnectivityMgr().GetAndLogWifiStatsCounters();
-    SuccessOrExit(err);
+    if (ConnectivityMgr().IsWiFiStationProvisioned() &&
+        ConnectivityMgr().IsWiFiStationEnabled() &&
+        ConfigurationMgr().IsPairedToAccount())
```
Comment (jaylogue) [inline src/adaptations/device-layer/NetworkTelemetryManager.cpp]: I'm going to accept this, but I'm still really uncertain about the IsPairedToAccount() logic.  Remind me the rationale for not logging stats when the device isn't paired to an account; in particular, what goes wrong if we do?

Comment (turon) [reply src/adaptations/device-layer/NetworkTelemetryManager.cpp]: @suryanshup I agree that capturing the answer to Jay's question this is important.

If the proper fix is being deferred, would it make sense to create a github issue with the explanation and reference it with a `//TODO(#issue)` comment?

Comment (suryanshup) [reply src/adaptations/device-layer/NetworkTelemetryManager.cpp]: Based on an offline conversation I had with Jay, we agreed that we should remove the service provisioning check. I will fix that in a separate PR
